File: python/images/py_028_generators/generators.py
def generator_numbers(numbers):
    for number in numbers:
        yield f"Number is {number}"


numbers = [1,2,3,4,5]
number_iterator=generator_numbers(numbers)
print(next(number_iterator))
# Number is 1
print(next(number_iterator))
# Number is 2

File: javascript/js_007_generators/generators.js
function* generateNumbers(numbers) {
  for (const number of numbers) {
    yield `Number is ${number}`;
  }
}

const numbers = [1, 2, 3, 4, 5];
const numberIterator = generateNumbers(numbers);

console.log(numberIterator.next());
// { value: 'Number is 1', done: false }
console.log(numberIterator.next());
// { value: 'Number is 2', done: false }
